packages/browser/src/sdk.ts

```diff
@@ -162,6 +162,9 @@ export function init(browserOptions: BrowserOptions = {}): Client | undefined {
     return;
   }
 
+  if (options.spotlight) {
```

Comment:
l: I wouldn't add this check. Browser options shouldn't have the `spotlight` property (do they?)

Reply:
No sorry we can't do this. There's no reason for the spotlight integration code to end up in a production build and it therefore needs to remain tree-shakeable. 

This is also why I'm asking to not include `spotlight: boolean` in the global options as IMHO this increases confusion more than it'd helps anyone. Btw, if you're worried about other non-node server runtimes, an alternative would be that we add the option to the global options but explicitly exclude it from the user-facing browser options. 
I'm fine with that as well. Our main goal should be that this option is not available in browser options.

RE printing the warning: I'd rather not have this:
- if you use TS, you get a type error anyway
- if you develop in plainJS, chances are high that your IDE will give you suggestions based on the type declarations. 
  - Sure not for everyone, I'm aware, but for a good part.
- There's no precedence for emitting warnings if users set an option that doesn't exist.

Reply:
>No sorry we can't do this. There's no reason for the spotlight integration code to end up in a production build and it therefore needs to remain tree-shakeable.

Is it possible for me to do a dynamic import if `spotlightBrowserIntegration` is not already in integrations then?

>if you use TS, you get a type error anyway

Sure, we ruled that out. That said it still doesn't fix the confusion if one expects this. The API between 2 JS SDKs become different for the very same task. From an end user perspective, this is just confusing.

>if you develop in plainJS, chances are high that your IDE will give you suggestions based on the type declarations.
>        Sure not for everyone, I'm aware, but for a good part.

I'll lump this into the one above as they are not really different.

>There's no precedence for emitting warnings if users set an option that doesn't exist.

Not having a precedence doesn't disqualify this from getting adopted. I do understand where you are coming from but if we are building Spotlight support into our SDKs, helping users with common errors become important for adoption and developer experience. Especially if the cost is not much.

Reply:
> it possible for me to do a dynamic import if spotlightBrowserIntegration is not already in integrations then?

I don't think this is easily doable in the browser SDK because it requires people to use a certain bundler config when they're bundling their code. So far we've abstained from dynamic imports for this reason. There's the alternative to use our `lazyloadIntegration` helper function but this also comes with bundle size implications because we still need to write that code 😅 Also, there's the whole timing issue with lazy loading because any events being sent before the integration loads lazily wouldn't make their way to spotlight.

Let's maybe approach it from this angle: Regardless of what we do, the question so far remained "should we add `spotlight: boolean` in the browser?". I would ask that - at least for now- we don't do this. Let's try to get get a self-contained integration exported from the browser SDK (so basically most of what this PR does :) ) and adapt our documentation for spotlight accordingly.

I fully understand you're worried about adoption but to the best of my knowledge, having worked on spotlight and the SKDs, I so far haven't seen issues where users are confused about why they needed to set `spotlight: true` in their backend SDKs but not in the browser SDK. If this comes up a lot in the future, and we really identify it as a major blocker for adoption, let's revisit what we can do. Sounds fair?

Btw, just that you don't think I'm deciding this all by myself, I discussed this with the team (actually multiple times) and we would like to proceed step-by-step here.

Reply:
Yeah, I'm okay going step by step. Was just pushing my luck regarding `spotlight: true`. Will revert the changes around that. That said for the following

>I fully understand you're worried about adoption but to the best of my knowledge, having worked on spotlight and the SKDs, I so far haven't seen issues where users are confused about why they needed to set spotlight: true in their backend SDKs but not in the browser SDK. If this comes up a lot in the future, and we really identify it as a major blocker for adoption, let's revisit what we can do. Sounds fair?

Be very careful with approaches like this as experience is actually a hinderence for this case (been into a similar situation). We are looking at people new at this. That said even myself who worked on and at Sentry for a while got _very_ confused with this. After that I did one user testing and they got confused about that too. The API dicrepancy is a real issue and needs to be addressed at some point if we want people to adopt Spotlight and distributed tracing which is where it really shines.